runtime/contract_update_validation.go

```diff
@@ -405,31 +397,51 @@ func (validator *ContractUpdateValidator) checkIdentifierEquality(
 func (validator *ContractUpdateValidator) checkConformances(
 	oldEnum *ast.CompositeDeclaration,
 	newEnum *ast.CompositeDeclaration,
-) error {
+) {
 
 	oldConformances := oldEnum.Conformances
 	newConformances := newEnum.Conformances
 
 	if len(oldConformances) != len(newConformances) {
-		return &ConformanceCountMismatchError{
+		validator.report(&ConformanceCountMismatchError{
 			expected: len(oldConformances),
 			found:    len(newConformances),
 			Range:    ast.NewRangeFromPositioned(newEnum.Identifier),
-		}
+		})
+
+		// If the lengths are not the same, trying to match the conformance
+		// may result in too many regression errors. hence return.
+		return
 	}
 
-	for index, conformance := range oldConformances {
+	for index, oldConformance := range oldConformances {
 		newConformance := newConformances[index]
-		err := conformance.CheckEqual(newConformance, validator)
+		err := oldConformance.CheckEqual(newConformance, validator)
 		if err != nil {
-			return &ConformanceMismatchError{
-				err:   err,
-				Range: ast.NewRangeFromPositioned(newConformance),
-			}
+			validator.report(&ConformanceMismatchError{
+				declName: newEnum.Identifier.Identifier,
+				err:      err,
+				Range:    ast.NewRangeFromPositioned(newConformance),
+			})
 		}
 	}
+}
 
-	return nil
+func (validator *ContractUpdateValidator) report(err error) {
+	if err == nil {
+		return
+	}
+	validator.errors = append(validator.errors, err)
+}
+
+func (validator *ContractUpdateValidator) getContractUpdateError() error {
+	return &ContractUpdateError{
+		contractName: validator.contractName,
+		Errors:       validator.errors,
+		LocationRange: interpreter.LocationRange{
```

Comment:
I think the range is missing here. Or maybe it should just be location directly here?

Reply:
Good point. We only have the location at this point. Updated in 28cb86c73633d88553a86f62dbcfb9dde72b5613